### ISSUE TITLE
getSunriseSunsetFunction is very slow.

### ISSUE THREAD

**Issue by spacekestrel** [aquilax/sunrise-sunset-calc#4]
I added this function to calculate sunrise and sunset on initState, but for some reason in takes appoximately 20 seconds to load app with it.

**Comment by aquilax** [comment]
Would you mind sharing some example code?

**Comment by spacekestrel** [comment]
Sure, but i haven't put it in the first place, because implementation is like in the example section. Here you go:
```  
void main() {
    SunriseSunsetResult sunriseSunsetResult = getSunriseSunset(
      lat: 56.0090,
      long: 37.8529,
      utcOffset: 3,
      dateTime: DateTime.now());
      sunriseGetxController.setSunrise(sunriseSunsetResult);
      runApp(Dashboard());
}
```
I am using get package to use data later.

**Comment by aquilax** [comment]
I added a benchmark harness in https://github.com/aquilax/sunrise-sunset-calc/pull/5 and running locally on i5 i am getting ~ `1.5s` per run, which is slow, but nowhere near the 20 seconds mark.

Can you double check if the `getSunriseSunset` is causing the 20 seconds delay? I guess it also makes a big difference on what cpu/architecture this is run.

**Comment by spacekestrel** [comment]
Maybe it worth mentioning that on emulator all works perfectly, but on real device - samsung galaxy s21  - it takes approximately 14 seconds. Upon launch screen is black for those 14 seconds and then app shows it's content. 
If i put this function in button -  app freeze on click event for the same amount of time.

**Comment by aquilax** [comment]
Thank you for reporting. 

Will try to take a look if there is any unnecessary slowdown in the code that can be optimized.

**Comment by spacekestrel** [comment]
Okay, thanks.

**Comment by aquilax** [comment]
@spacekestrel i just released `v2.1.0`, which shows faster times in the benchmark:
```
$ dart run benchmark/sunrise_sunset_calc_benchmark.dart 
SunriseSunset(RunTime): 1129117.5 us.
```

**Comment by spacekestrel** [comment]
@aquilax Tested the update . The download time has decreased to about 6-8 seconds. That is, it became faster, but it is still too long, given that the Galaxy S21 has the most powerful android processor at the moment. 
Have you seen this algorithm?
https://web.archive.org/web/20161202180207/http://williams.best.vwh.net/sunrise_sunset_algorithm.htm
I took it from the daylight package.
